Doc/whatsnew/3.11.rst

```diff
@@ -1884,6 +1884,17 @@ New Features
 Porting to Python 3.11
 ----------------------
 
+* Many macros have been converted to functions in the Python C API,
+  making them usable from other programming languages and avoiding
+  `macro pitfalls <https://gcc.gnu.org/onlinedocs/cpp/Macro-Pitfalls.html>`_
```

Comment:
```suggestion
* Some macros have been converted to static inline functions to avoid
  `macro pitfalls <https://gcc.gnu.org/onlinedocs/cpp/Macro-Pitfalls.html>`_.
```

* Changed "many" to "some" and removed "making them usable from other programming languages" per @vstinner 
* Added "static inline" per @kumaraditya303 
* Elided "in the C API", since this is under the C API section, and it concerns the ABI as well as the API